### ISSUE TITLE
Default-812h@3x~iphone.png and Default-Landscape-812h@3x~iphone.png

### ISSUE THREAD

**Issue by ssj7** [harshzalavadiya/ionic-resource-generator#5]
Hi,

My icon.png is 1024x1024 and splash.png is 2732x2732

O̶n̶e̶ ̶i̶c̶o̶n̶ ̶a̶n̶d̶ ̶ 2 splash screens are always missing when generating:

**ios**

-̶ ̶i̶c̶o̶n̶s̶

`I̶c̶o̶n̶-̶2̶0̶.̶p̶n̶g̶ ̶(̶2̶0̶x̶2̶0̶)̶`

- splash

```
Default-812h@3x~iphone.png (1125x2436)
Default-Landscape-812h@3x~iphone.png (2436x1125)
```
Edit: The icon Icon-20.png (20x20) gets generated by this tool, only those 2 splash screens are missing(watch and tvOS icons also but am not using those).

**Comment by harshzalavadiya** [comment]
can you provide minimal ionic project so i can debug the issue

**Comment by ssj7** [comment]
`ionic start NewApp blank --cordova --type=ionic-angular`
Fresh ionic v3 project. If you want I can upload the resources folder.

**Comment by harshzalavadiya** [comment]
Thanks, I'll look into issue
but for ionic you don't need this project as of now
ionic team officially provides support for offline thumbnail generation

https://github.com/ionic-team/cordova-res

**Comment by ssj7** [comment]
Yes in most cases and fresh installations, but for most of us who have to do `npm audit fix` after installing many packages or when updating projects/plugins stuff gets broken and your solution works very well and is so simple to not use it :)

I have seen a case where an official ionic dev/admin recommended falling back to online ionic upload resource gen( `ionic cordova resources  --no-cordova-res`) when cordova-res had conflict with some npm packages that required a lot of other installations  and security audits to fix(
`npm install expo-cli -g` `npm install sharp`), while your solution fixes it in a matter of seconds. There is also an issue with cordova-res on mac (last time I checked, probably on my end or a mac thing).

Thanks a lot for this, I can generate the missing icon and 2 splash screens manually, it doesn't affect me very much, but I wanted to let you know in case you wanted to keep it up to date.

**Comment by harshzalavadiya** [comment]
Hi @ssj7 ,

Thanks for the feedback :smile: 

I am happy that this project is useful to someone and yes I always wanted to keep it up to date

I tried creating project with latest ionic CLI and tried generating resources, i was able to generate resources properly but there was no image with `Default-812h@3x~iphone.png` and `Default-Landscape-812h@3x~iphone.png`

Attaching reference resources folder

[resources.zip](https://github.com/harshzalavadiya/ionic-resource-generator/files/3508437/resources.zip)

Basically this utility iterates through android and ios folder and updates all icon and splash folder png's appropriately if it's outside of that folder it won't generate

If possible can you provide sample resources folder?

**Comment by ssj7** [comment]
Resources folder with ` Default-812h@3x~iphone.png and Default-Landscape-812h@3x~iphone.png ` splash screens for iOS. Sorry for my late replies. I can also make another resources folder with icons/splash for tvOS and watchOS if you want. 

[resources.zip](https://github.com/harshzalavadiya/ionic-resource-generator/files/3513409/resources.zip)

**Comment by harshzalavadiya** [comment]
Thank you for the resources,
if they are not generating as well then please do so

**Comment by harshzalavadiya** [comment]
Hi @ssj7,
Sorry for late response
I tried running it locally and binary was able to transform all images properly on my Linux machine, sorry I don't have apple hardware to test on as of now.
but as of now I can't do anything about this so I'm going to keep this open and see what I can do about this in future

**Comment by ssj7** [comment]
Hello,
it has been a while since I last updated an ionic/cordova app, but the last time I did 
```
ionic cordova resources --force
ionic cordova resources  --no-cordova-res
```
Worked splendidly.
